queue/group.go

```diff
@@ -0,0 +1,132 @@
+package queue
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/mongodb/amboy"
+	"github.com/pkg/errors"
+)
+
+// NewLocalQueueGroup constructs a new local queue group. If ttl is 0, the collection will not be
+// TTLed except when the client explicitly calls Prune.
+func NewLocalQueueGroup(ctx context.Context, constructor QueueConstructor, ttl time.Duration) (amboy.QueueGroup, error) {
+	if constructor == nil {
+		return nil, errors.New("must pass a constructor")
+	}
+	if ttl < 0 {
+		return nil, errors.New("ttl must be greater than or equal to 0")
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	g := &localQueueGroup{
+		canceler:    cancel,
+		queues:      map[string]amboy.Queue{},
+		constructor: constructor,
+		ttlMap:      map[string]time.Time{},
+		ttl:         ttl,
+	}
+	if ttl > 0 {
+		go func(ctx context.Context) {
```

Comment:
recovery handler. 

also I don't think you need to pass context here because it's in the outer scope